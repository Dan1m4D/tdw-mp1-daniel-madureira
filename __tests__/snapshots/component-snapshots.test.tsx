import React from "react";
import { render } from "@testing-library/react";
import Avatar from "@/app/avatar";
import CoverImage from "@/app/cover-image";
import MoreStories from "@/app/more-stories";
import type { Post } from "@/lib/types";

describe("Component Snapshots", () => {
  it("Avatar matches snapshot", () => {
    const { container } = render(
      <Avatar
        name="John Doe"
        picture={{ url: "https://example.com/avatar.jpg" }}
      />,
    );
    expect(container).toMatchSnapshot();
  });

  it("CoverImage matches snapshot with link", () => {
    const { container } = render(
      <CoverImage
        title="Blog Post Title"
        url="https://example.com/cover.jpg"
        slug="blog-post-slug"
      />,
    );
    expect(container).toMatchSnapshot();
  });

  it("MoreStories matches snapshot with posts", () => {
    const mockPosts: Post[] = [
      {
        title: "First Post",
        slug: "first-post",
        date: "2023-01-15",
        excerpt: "This is the first post excerpt",
        coverImage: { url: "https://example.com/post1.jpg" },
        author: {
          name: "Author One",
          picture: { url: "https://example.com/author1.jpg" },
        },
      },
      {
        title: "Second Post",
        slug: "second-post",
        date: "2023-02-20",
        excerpt: "This is the second post excerpt",
        coverImage: { url: "https://example.com/post2.jpg" },
        author: {
          name: "Author Two",
          picture: { url: "https://example.com/author2.jpg" },
        },
      },
    ];

    const { container } = render(<MoreStories morePosts={mockPosts} />);
    expect(container).toMatchSnapshot();
  });
});
