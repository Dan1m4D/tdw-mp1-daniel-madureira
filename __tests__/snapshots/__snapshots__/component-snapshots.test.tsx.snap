// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Component Snapshots Avatar matches snapshot 1`] = `
<div>
  <div
    class="flex items-center"
  >
    <div
      class="mr-4 w-12 h-12"
    >
      <img
        alt="John Doe"
        class="object-cover h-full rounded-full"
        data-nimg="1"
        decoding="async"
        height="48"
        loading="lazy"
        src="https://example.com/avatar.jpg?w=96&q=75"
        srcset="https://example.com/avatar.jpg?w=48&q=75 1x, https://example.com/avatar.jpg?w=96&q=75 2x"
        style="color: transparent;"
        width="48"
      />
    </div>
    <div
      class="text-xl font-bold"
    >
      John Doe
    </div>
  </div>
</div>
`;

exports[`Component Snapshots CoverImage matches snapshot with link 1`] = `
<div>
  <div
    class="sm:mx-0"
  >
    <a
      aria-label="Blog Post Title"
      href="/posts/blog-post-slug"
    >
      <img
        alt="Cover Image for Blog Post Title"
        class="shadow-small [object Object]"
        data-nimg="1"
        decoding="async"
        height="1000"
        src="https://example.com/cover.jpg?w=3840&q=75"
        srcset="https://example.com/cover.jpg?w=2048&q=75 1x, https://example.com/cover.jpg?w=3840&q=75 2x"
        style="color: transparent;"
        width="2000"
      />
    </a>
  </div>
</div>
`;

exports[`Component Snapshots MoreStories matches snapshot with posts 1`] = `
<div>
  <section>
    <h2
      class="mb-8 text-6xl md:text-7xl font-bold tracking-tighter leading-tight"
    >
      More Stories
    </h2>
    <div
      class="grid grid-cols-1 md:grid-cols-2 md:gap-x-16 lg:gap-x-32 gap-y-20 md:gap-y-32 mb-32"
    >
      <div>
        <div
          class="mb-5"
        >
          <div
            class="sm:mx-0"
          >
            <a
              aria-label="First Post"
              href="/posts/first-post"
            >
              <img
                alt="Cover Image for First Post"
                class="shadow-small [object Object]"
                data-nimg="1"
                decoding="async"
                height="1000"
                src="https://example.com/post1.jpg?w=3840&q=75"
                srcset="https://example.com/post1.jpg?w=2048&q=75 1x, https://example.com/post1.jpg?w=3840&q=75 2x"
                style="color: transparent;"
                width="2000"
              />
            </a>
          </div>
        </div>
        <h3
          class="text-3xl mb-3 leading-snug"
        >
          <a
            class="hover:underline"
            href="/posts/first-post"
          >
            First Post
          </a>
        </h3>
        <div
          class="text-lg mb-4"
        >
          <time
            datetime="2023-01-15"
          >
            January 15, 2023
          </time>
        </div>
        <p
          class="text-lg leading-relaxed mb-4"
        >
          This is the first post excerpt
        </p>
        <div
          class="flex items-center"
        >
          <div
            class="mr-4 w-12 h-12"
          >
            <img
              alt="Author One"
              class="object-cover h-full rounded-full"
              data-nimg="1"
              decoding="async"
              height="48"
              loading="lazy"
              src="https://example.com/author1.jpg?w=96&q=75"
              srcset="https://example.com/author1.jpg?w=48&q=75 1x, https://example.com/author1.jpg?w=96&q=75 2x"
              style="color: transparent;"
              width="48"
            />
          </div>
          <div
            class="text-xl font-bold"
          >
            Author One
          </div>
        </div>
      </div>
      <div>
        <div
          class="mb-5"
        >
          <div
            class="sm:mx-0"
          >
            <a
              aria-label="Second Post"
              href="/posts/second-post"
            >
              <img
                alt="Cover Image for Second Post"
                class="shadow-small [object Object]"
                data-nimg="1"
                decoding="async"
                height="1000"
                src="https://example.com/post2.jpg?w=3840&q=75"
                srcset="https://example.com/post2.jpg?w=2048&q=75 1x, https://example.com/post2.jpg?w=3840&q=75 2x"
                style="color: transparent;"
                width="2000"
              />
            </a>
          </div>
        </div>
        <h3
          class="text-3xl mb-3 leading-snug"
        >
          <a
            class="hover:underline"
            href="/posts/second-post"
          >
            Second Post
          </a>
        </h3>
        <div
          class="text-lg mb-4"
        >
          <time
            datetime="2023-02-20"
          >
            February 20, 2023
          </time>
        </div>
        <p
          class="text-lg leading-relaxed mb-4"
        >
          This is the second post excerpt
        </p>
        <div
          class="flex items-center"
        >
          <div
            class="mr-4 w-12 h-12"
          >
            <img
              alt="Author Two"
              class="object-cover h-full rounded-full"
              data-nimg="1"
              decoding="async"
              height="48"
              loading="lazy"
              src="https://example.com/author2.jpg?w=96&q=75"
              srcset="https://example.com/author2.jpg?w=48&q=75 1x, https://example.com/author2.jpg?w=96&q=75 2x"
              style="color: transparent;"
              width="48"
            />
          </div>
          <div
            class="text-xl font-bold"
          >
            Author Two
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
